blocks/tabs/tabs.js

```diff
@@ -0,0 +1,285 @@
+    switch (tabId.toLowerCase()) {
+      case 'videos':
+        textContent.appendChild(createCardMetaElement(item.date, item.author));
+        break;
+      case 'podcasts':
+        textContent.appendChild(createCardMetaElement(item.date, item.duration));
+        break;
+      default:
+        break;
+    }
+    textContent.appendChild(createPostTitle(item));
+    textContent.appendChild(createSocialLinkElement(item));
+
+    // Append picture-wrapper and text-content to cardInfo
+    cardInfo.appendChild(mediaWrapper);
+    cardInfo.appendChild(textContent);
+
+    // Append cardInfo to slide
+    card.appendChild(cardInfo);
+
+    // Append slide to container
+    container.appendChild(card);
+  });
+}
+
+function createDots(totalCards, maxAllowedCards, dots) {
+  const numberOfDotsToBeCreated = numberOfDots(totalCards, maxAllowedCards);
+  let index = 1;
+  while (index <= numberOfDotsToBeCreated) {
+    const dot = document.createElement('button');
+    dot.className = 'dot';
+    dot.dataset.index = index;
+    dots.appendChild(dot);
+    dot.addEventListener('click', (event) => {
+      targetedDotView(event);
+    });
+    index += 1;
+  }
+  dots.firstElementChild.classList.add('active');
+}
+
+async function createTabPanel(block) {
+  const tabsPanel = block.querySelectorAll('.tabs-panel');
+  const promises = [];
+
+  for (let index = 0; index < tabsPanel.length; index += 1) {
+    const tab = tabsPanel[index];
+    const discoverMoreButton = tab.lastElementChild;
+    tab.textContent = '';
+    const tabId = tab.getAttribute('api-key');
+    const title = document.createElement('div');
+    title.className = 'title';
+    const h2 = document.createElement('h2');
+    const img = document.createElement('img');
+    img.src = '/icons/video-icon.png';
+    const picture = document.createElement('picture');
+    picture.appendChild(img);
+    const textNode = document.createTextNode(tabId);
+    h2.appendChild(picture);
+    h2.appendChild(textNode);
+    title.append(h2);
+    tab.appendChild(title);
+    const slider = document.createElement('div');
+    slider.className = 'slider';
+    tab.appendChild(slider);
+    const track = document.createElement('div');
+    track.className = 'track';
+    const dots = document.createElement('div');
+    dots.className = 'dots-container';
+    slider.appendChild(track);
+    slider.appendChild(dots);
+    discoverMoreButton.className = 'discovermore';
+    if (discoverMoreButton.firstElementChild && discoverMoreButton.firstElementChild.tagName === 'A') {
+      discoverMoreButton.firstElementChild.className = 'discover-more-button';
+    }
+    tab.appendChild(discoverMoreButton);
+
+    // Collect promises from asynchronous operations
+    promises.push(getTabDataAPI(tabId).then((data) => {
+      createCards(track, data, tabId);
+      createDots(data.length, allowedCardsCount(), dots);
+    }));
+  }
+  // Wait for all promises to resolve
+  await Promise.all(promises);
```

Comment:
Please see [this comment](https://github.com/aemsites/icicidirect/pull/82/files#r1548731220). I think the same applies here.